src/main/java/com/google/sps/classes/Utility.java

```diff
@@ -115,4 +123,80 @@ public static void addNewUser(String firstName, String lastName, String username
+        ResultSet rs = pst.executeQuery()) {
+        rs.next();
+        // Concatenate the user's email and a comma for the InternetAddress parser to separate.
+        userEmails = userEmails.concat(rs.getString(1));
+        userEmails = userEmails.concat(",");
+        connection.close();
+      } catch (SQLException ex) {
+        Logger lgr = Logger.getLogger(EmailServlet.class.getName());
+        lgr.log(Level.SEVERE, ex.getMessage(), ex);
+      }
+    }
+    // Erase the last comma.
+    userEmails = userEmails.substring(0, userEmails.length() - 1);
+    return userEmails;
+  }
+
+  /**
+   * Queries IDs of the author of the modified question/answer and its followers. Includes url, 
+   * user and password to give the user the ability to choose between local and cloud SQL 
+   * variables.
+   */
+  private List<Integer> getUsersToNotify(String typeOfNotification, int modifiedElementId,
+                                         String SQL_URL, String SQL_USER, String SQL_PASSWORD) {
+    List<Integer> usersToNotify = new ArrayList<>();
+    if(typeOfNotification.equals("question")) {
+      // If the notification is for an anwer to a question.
+      String query =  "SELECT follower_id FROM QuestionFollower WHERE question_id = " +
+                      modifiedElementId;
+      // Query the information from QuestionFollower table.
+      try (Connection connection = DriverManager.getConnection(SQL_URL, SQL_USER, SQL_PASSWORD);
+          PreparedStatement pst = connection.prepareStatement(query);
+          ResultSet rs = pst.executeQuery()) {
+        while(rs.next()){
+          // Add the current ID (first column of ResultSet) to the list.
+          usersToNotify.add(rs.getInt(1));
+        }
+        // Close the connection once the query was performed have been performed.
+        connection.close();
+      } catch (SQLException ex) {
+        Logger lgr = Logger.getLogger(DataServlet.class.getName());
+        lgr.log(Level.SEVERE, ex.getMessage(), ex);
+      }
+    } else if (typeOfNotification.equals("answer")) {
+      // If the notification is for a new comment in an answer.
+      String query =  "SELECT follower_id FROM AnswerFollower WHERE answer_id = " +
+                      modifiedElementId;
+      // Query the information from AnswerFollower table.
+      try (Connection connection = DriverManager.getConnection(SQL_URL, SQL_USER, SQL_PASSWORD);
+           PreparedStatement pst = connection.prepareStatement(query);
+           ResultSet rs = pst.executeQuery()) {
+        while(rs.next()){
+          // Add the current ID (first column of ResultSet) to the list.
+          usersToNotify.add(rs.getInt(1));
+        }
+        // Close the connection once the query was performed have been performed.
+        connection.close();
+      } catch (SQLException ex) {
+          Logger lgr = Logger.getLogger(EmailServlet.class.getName());
+          lgr.log(Level.SEVERE, ex.getMessage(), ex);
+      }
+    }
```

Comment:
You could condense this by combining the logic and doing something similar to this:

```suggestion
    string query = "";
    if (typeOfNotification.equals("question")) {
      query =  "SELECT follower_id FROM QuestionFollower WHERE question_id = " +
                      modifiedElementId;    
    } else if (typeOfNotification.equals("answer")) {
      query =  "SELECT follower_id FROM AnswerFollower WHERE answer_id = " +
                      modifiedElementId;
    }
    
    if (query == "") { return usersToNotify;}
    
    try (Connection connection = DriverManager.getConnection(SQL_URL, SQL_USER, SQL_PASSWORD);
           PreparedStatement pst = connection.prepareStatement(query);
           ResultSet rs = pst.executeQuery()) {
        while(rs.next()){
          // Add the current ID (first column of ResultSet) to the list.
          usersToNotify.add(rs.getInt(1));
        }
        // Close the connection once the query was performed have been performed.
        connection.close();
      } catch (SQLException ex) {
          Logger lgr = Logger.getLogger(EmailServlet.class.getName());
          lgr.log(Level.SEVERE, ex.getMessage(), ex);
      }
    }
```

Reply:
Thanks for the suggestion, done!